src/directive/sort-by.directive.ts

```diff
@@ -53,11 +65,23 @@ export class JhiSortByDirective implements AfterViewInit {
     }
 
     private applyClass() {
-        const childSpan = this.el.nativeElement.children[1];
-        let add = this.sortAscIcon;
-        if (!this.jhiSort.ascending) {
-            add = this.sortDescIcon;
+        let icon: any = this.sortIcon;
+        let iconName = this.sortIconName;
+        if (this.jhiSort.predicate === this.jhiSortBy) {
+            if (this.jhiSort.ascending) {
+                icon = this.sortAscIcon;
+                iconName = this.sortAscIconName;
+            } else {
+                icon = this.sortDescIcon;
+                iconName = this.sortDescIconName;
+            }
         }
-        this.renderer.setElementClass(childSpan, add, true);
-    };
+
+        const childIcon = this.el.nativeElement.children[1];
+        this.renderer.setElementAttribute(childIcon.children[0], 'data-icon', iconName);
+        this.renderer.setElementAttribute(childIcon.children[0], 'class', 'svg-inline--fa fa-w-10 ' + iconName);
+        // update the path with de 5th icon attribute.
+        //
+        this.renderer.setElementAttribute(childIcon.children[0].children[0], 'd', icon.definition.icon[4]);
```

Comment:
can we somehow make this better? It is too hacky

Reply:
I must use the css selector like the precedent post. For the path, it’s The 5th element of an array. There is no properties.

Reply:
I must use the css selector like the precedent post. For the path, it’s The 5th element of an array. There is no properties.